server/data_structures/internal_datafacade.hpp

```diff
@@ -258,90 +257,40 @@ template <class EdgeDataT> class InternalDataFacade final : public BaseDataFacad
+        };
+
+        ram_index_path = file_for("ramindex");
+        file_index_path = file_for("fileindex");
 
-        auto paths_iterator = server_paths.find("hsgrdata");
-        BOOST_ASSERT(server_paths.end() != paths_iterator);
-        const boost::filesystem::path &hsgr_path = paths_iterator->second;
-        paths_iterator = server_paths.find("timestamp");
-        BOOST_ASSERT(server_paths.end() != paths_iterator);
-        const boost::filesystem::path &timestamp_path = paths_iterator->second;
-        paths_iterator = server_paths.find("ramindex");
-        BOOST_ASSERT(server_paths.end() != paths_iterator);
-        ram_index_path = paths_iterator->second;
-        paths_iterator = server_paths.find("fileindex");
-        BOOST_ASSERT(server_paths.end() != paths_iterator);
-        file_index_path = paths_iterator->second;
-        paths_iterator = server_paths.find("nodesdata");
-        BOOST_ASSERT(server_paths.end() != paths_iterator);
-        const boost::filesystem::path &nodes_data_path = paths_iterator->second;
-        paths_iterator = server_paths.find("edgesdata");
-        BOOST_ASSERT(server_paths.end() != paths_iterator);
-        const boost::filesystem::path &edges_data_path = paths_iterator->second;
-        paths_iterator = server_paths.find("namesdata");
-        BOOST_ASSERT(server_paths.end() != paths_iterator);
-        const boost::filesystem::path &names_data_path = paths_iterator->second;
-        paths_iterator = server_paths.find("geometries");
-        BOOST_ASSERT(server_paths.end() != paths_iterator);
-        const boost::filesystem::path &geometries_path = paths_iterator->second;
-        paths_iterator = server_paths.find("coredata");
-        BOOST_ASSERT(server_paths.end() != paths_iterator);
-        const boost::filesystem::path &core_data_path = paths_iterator->second;
-
-        // load data
         SimpleLogger().Write() << "loading graph data";
-        AssertPathExists(hsgr_path);
-        LoadGraph(hsgr_path);
+        LoadGraph(file_for("hsgrdata"));
+
         SimpleLogger().Write() << "loading edge information";
-        AssertPathExists(nodes_data_path);
-        AssertPathExists(edges_data_path);
-        LoadNodeAndEdgeInformation(nodes_data_path, edges_data_path);
+        LoadNodeAndEdgeInformation(file_for("nodesdata"), file_for("edgesdata"));
+
         SimpleLogger().Write() << "loading core information";
-        AssertPathExists(core_data_path);
-        LoadCoreInformation(core_data_path);
+        LoadCoreInformation(file_for("coredata"));
+
         SimpleLogger().Write() << "loading geometries";
-        AssertPathExists(geometries_path);
-        LoadGeometries(geometries_path);
+        LoadGeometries(file_for("geometries"));
+
         SimpleLogger().Write() << "loading r-tree";
-        AssertPathExists(ram_index_path);
-        AssertPathExists(file_index_path);
+        // XXX(daniel-j-h): it says this ^ but doesn't load the rtree here
```

Comment:
The `StaticRTree` is a thread specific pointer that is initialized on demand the first time it is used. It needs to be thread specific because it contains a stream object which is not thread safe. Which make me think we should maybe explore `mmap`, since virtual memory accesses is thread safe.